graylog2-server/src/test/java/org/graylog/testing/mongodb/MongoDBFixtureImporter.java

```diff
@@ -0,0 +1,243 @@
+        return Arrays.stream(resourceNames)
+                .map(resourceName -> toResource(resourceName, contextClass))
+                .collect(Collectors.toList());
+    }
+
+    private static URL toResource(final String resourceName, final Class<?> contextClass) {
+        if (Paths.get(resourceName).getNameCount() > 1) {
+            return Resources.getResource(resourceName);
+        } else {
+            return Resources.getResource(contextClass, resourceName);
+        }
+    }
+
+    private void importResource(URL resource, MongoDatabase database) {
+        LOG.debug("Importing fixture resource: {}", resource);
+        try {
+            importData(database, Resources.toString(resource, StandardCharsets.UTF_8));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void importData(MongoDatabase database, String jsonData) {
+        final Document data = Document.parse(jsonData);
+
+        for (final String collectionName : data.keySet()) {
+            final Object document = data.get(collectionName, Object.class);
+
+            if (isListOfDocuments(document)) {
+                importDocuments(database, collectionName, data.get(collectionName, List.class));
+            } else {
+                final Document doc = (Document) document;
+                importIndexes(database, collectionName, doc);
+                importDocuments(database, collectionName, doc.get(FIELD_DATA, List.class));
+            }
+        }
+    }
+
+    private boolean isListOfDocuments(Object object) {
+        return List.class.isAssignableFrom(object.getClass());
+    }
+
+    private void importDocuments(MongoDatabase database, String collectionName, List<Document> dataObjects) {
+        final MongoCollection<Document> collection = database.getCollection(collectionName);
+
+        for (final Document dataObject : dataObjects) {
+            collection.insertOne(dataObject);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void importIndexes(MongoDatabase database, String collectionName, Document doc) {
+        if (!doc.containsKey(FIELD_INDEXES)) {
+            return;
+        }
+
+        final MongoCollection<Document> collection = database.getCollection(collectionName);
+        final List<Document> indexes = doc.get(FIELD_INDEXES, List.class);
+
+        for (final Document indexDoc : indexes) {
+            final Document indexFields = indexDoc.get(FIELD_INDEX, Document.class);
+
+            if (indexDoc.containsKey(FIELD_INDEX_OPTIONS)) {
+                collection.createIndex(indexFields, createIndexOptions(indexDoc.get(FIELD_INDEX_OPTIONS, Document.class)));
+            } else {
+                collection.createIndex(indexFields);
+            }
+        }
+    }
+
+    private IndexOptions createIndexOptions(Document indexOptionsDoc) {
```

Comment:
I assume you have left out [`collation`](https://docs.mongodb.com/manual/reference/bson-type-comparison-order/#collation) on purpose, because we don't need it, right?

Reply:
Yes, I think we can add that once we need it. :smiley: